src/python/pants/rules/core/repl.py

```diff
@@ -56,8 +70,21 @@ class ReplBinary:
     # We can guarantee that we will only even enter this `goal_rule` if there exists an implementer
     # of the `ReplImplementation` union because `LegacyGraphSession.run_goal_rules()` will not
     # execute this rule's body if there are no implementations registered.
-    repl_impl = next(iter(union_membership.union_rules[ReplImplementation]))
-    repl_binary = await Get[ReplBinary](ReplImplementation, repl_impl(addresses))
+    membership = union_membership.union_rules[ReplImplementation]
+    implementations = {impl.name: impl for impl in membership}
+
+    default_repl = "python"
+    repl_shell_name: str = options.values.shell or default_repl
+
+    repl_implementation = implementations.get(repl_shell_name)
+    if repl_implementation is None:
+        available = set(implementations.keys())
+        console.write_stdout(
+            f"{repl_shell_name} is not an installed REPL program. Available REPLs: {available}"
+        )
+        return Repl(-1)
```

Comment:
The `-1` is to signal that this was a Pants error and not an error from the underlying REPL implementation?

Reply:
That's the idea. I'm not sure if -1 is the best way to signal this, but it seems like it shouldn't return a success 0.